controller/controller_test.go

```diff
@@ -968,13 +968,10 @@ func TestStartAndShutdownWithLeaderAwareNoElection(t *testing.T) {
 	})
 
 	select {
-	case <-promoted:
-		// We expect to be promoted immediately, since there is no
-		// ElectorBuilder attached to the context.
 	case <-doneCh:
 		t.Fatal("StartAll finished early.")
 	case <-time.After(10 * time.Second):
-		t.Error("Timed out waiting for StartAll.")
+		// Give it some time to run.
```

Comment:
the loss of the error message might be a problem? This will fall through to the cancel at line 977 and you might not understand if or why it timed out.

Reply:
This is now the normal path for the test.  Previously `PromoteFunc` was called, which triggered another path.  Now this is waiting to effectively make sure it isn't called.